letsencrypt-compatibility-test/letsencrypt_compatibility_test/configurators/apache/common.py

```diff
@@ -0,0 +1,227 @@
+        subprocess_patch = mock.patch(
+            "letsencrypt_apache.configurator.subprocess")
+        subprocess_mock = subprocess_patch.start()
+        subprocess_mock.check_call = self.check_call_in_docker
+        subprocess_mock.Popen = self.popen_in_docker
+        self._patches.append(subprocess_patch)
+
+        display_patch = mock.patch(
+            "letsencrypt_apache.configurator.display_ops.select_vhost")
+        display_mock = display_patch.start()
+        display_mock.side_effect = le_errors.PluginError(
+            "Unable to determine vhost")
+        self._patches.append(display_mock)
+
+        self.modules = self.server_root = self.test_conf = self.version = None
+        self._apache_configurator = self._all_names = self._test_names = None
+
+    def __getattr__(self, name):
+        """Wraps the Apache Configurator methods"""
+        method = getattr(self._apache_configurator, name, None)
+        if callable(method):
+            return method
+        else:
+            raise AttributeError()
+
+    def load_config(self):
+        """Loads the next configuration for the plugin to test"""
+        if hasattr(self.le_config, "apache_init_script"):
+            try:
+                self.check_call_in_docker(
+                    [self.le_config.apache_init_script, "stop"])
+            except errors.Error:
+                raise errors.Error(
+                    "Failed to stop previous apache config from running")
+
+        config = super(Proxy, self).load_config()
+        self.modules = _get_modules(config)
+        self.version = _get_version(config)
+        self._all_names, self._test_names = _get_names(config)
+
+        server_root = _get_server_root(config)
+        with open(os.path.join(config, "config_file")) as f:
+            config_file = os.path.join(server_root, f.readline().rstrip())
+        self.test_conf = _create_test_conf(server_root, config_file)
+
+        self.preprocess_config(server_root)
+        self._prepare_configurator(server_root, config_file)
+
+        try:
+            self.check_call_in_docker(
+                "apachectl -d {0} -f {1} -k start".format(
+                    server_root, config_file))
+        except errors.Error:
+            raise errors.Error(
+                "Apache failed to load {0} before tests started".format(
+                    config))
+
+        return config
+
+    def preprocess_config(self, server_root):
+        # pylint: disable=anomalous-backslash-in-string, no-self-use
+        """Prepares the configuration for use in the Docker"""
+
+        find = subprocess.Popen(
+            ["find", server_root, "-type", "f"],
+            stdout=subprocess.PIPE)
+        subprocess.check_call([
+            "xargs", "sed", "-e", "s/DocumentRoot.*/DocumentRoot "
+            "\/usr\/local\/apache2\/htdocs/I",
+            "-e", "s/SSLPassPhraseDialog.*/SSLPassPhraseDialog builtin/I",
+            "-e", "s/TypesConfig.*/TypesConfig "
```

Comment:
The substitution is case insensitive. The expression is just continued on the next line due to length.